### PR TITLE
`if` chains → `switch` expressions

### LINKED ISSUE #144903: refactor `if` chains and `==` with enum values
From the [Flutter repo style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values):

> ### Avoid using `if` chains or `?:` or `==` with enum values

<br>

#### Example (toggle_buttons.dart):

```dart
// using "=="
bool _isFirstButton(int index, int length, TextDirection textDirection) {
  return index == 0 && ((direction == Axis.horizontal && textDirection == TextDirection.ltr) ||
    (direction == Axis.vertical && verticalDirection == VerticalDirection.down))
    || index == length - 1 && ((direction == Axis.horizontal && textDirection == TextDirection.rtl) ||
    (direction == Axis.vertical && verticalDirection == VerticalDirection.up));
}


// using a switch statement
bool _isFirstButton(int index, int length, TextDirection textDirection) {
  switch (direction) {
    case Axis.horizontal:
      return switch (textDirection) {
        TextDirection.rtl => index == length - 1,
        TextDirection.ltr => index == 0,
      };
    case Axis.vertical:
      return switch (verticalDirection) {
        VerticalDirection.up   => index == length - 1,
        VerticalDirection.down => index == 0,
      };
  }
}
```

Performing this refactor would increase the total line length, but in my opinion the improved readability is worth it.

### PR DESCRIPTION
**Pull request by nate-thegrate**
Previous "if-chains" pull requests:
- https://github.com/flutter/flutter/pull/144905
- https://github.com/flutter/flutter/pull/144977
- https://github.com/flutter/flutter/pull/145194
- https://github.com/flutter/flutter/pull/146293
- https://github.com/flutter/flutter/pull/147472

<br>

I think this one should be enough to wrap things up!

fixes https://github.com/flutter/flutter/issues/144903

### CODE REVIEW

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -914,12 +914,10 @@ abstract class ResidentHandlers {
...
       isolateId: views.first.uiIsolate!.id!,
     );
-    Brightness next;
-    if (current == Brightness.light) {
-      next = Brightness.dark;
-    } else {
-      next = Brightness.light;
-    }
+    final Brightness next = switch (current) {
+      Brightness.light || null => Brightness.dark,
```
Comment (victorsanni) [inline packages/flutter_tools/lib/src/resident_runner.dart:918]: My understanding of the if statement is that if and only if `current == Brightness.light` should `next = Brightness.dark`, otherwise (including the `null` case) `next = Brightness.light`. Is this change intended?

File: packages/flutter_tools/test/src/context.dart [hunk 2]
```diff
@@ -290,13 +290,11 @@ class FakeDeviceManager implements DeviceManager {
...
-      return attachedDevices;
-    }
-    if (filter?.deviceConnectionInterface == DeviceConnectionInterface.wireless) {
-      return wirelessDevices;
-    }
-    return attachedDevices + wirelessDevices;
+    return switch (filter?.deviceConnectionInterface) {
+      DeviceConnectionInterface.attached => attachedDevices,
+      DeviceConnectionInterface.wireless => wirelessDevices,
+      null => attachedDevices + wirelessDevices,
```
Comment (victorsanni) [inline packages/flutter_tools/test/src/context.dart:296]: I'm a little curious here as to why `null` is used instead of an arbitrary `_` as in (almost) everywhere else.

**Review by victorsanni — APPROVED** [review]
Overall LGTM! I just some small questions that I wasn't too sure about.

File: packages/flutter_tools/lib/src/resident_runner.dart (hunk 1 above)
Comment (nate-thegrate) [reply packages/flutter_tools/lib/src/resident_runner.dart:918]: Good catch, thank you!

The `|| null` should go next to `Brightness.dark`.

File: packages/flutter_tools/test/src/context.dart (hunk 2 above)
Comment (nate-thegrate) [reply packages/flutter_tools/test/src/context.dart:296]: `Enum`, `bool`, and `sealed` classes are exhaustive—Dart is able to understand when all cases have been covered.

A wildcard pattern `_` in a switch expression (or `default` in a switch statement) goes against the [style guideline](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values) that this pull request was named after.

I made this mistake during my first [switch expressions PR](https://github.com/flutter/flutter/pull/136140#discussion_r1349586844), but never again. 🙂

**Comment by nate-thegrate** [comment]
Merging manually, since `autosubmit` bot is down.
